redash/query_runner/mysql.py

```diff
@@ -140,21 +140,39 @@ def run_query(self, query, user):
                                          ssl=self._get_ssl_parameters())
             cursor = connection.cursor()
             logger.debug("MySQL running query: %s", query)
-            cursor.execute(query)
-
-            data = cursor.fetchall()
-
-            # TODO - very similar to pg.py
-            if cursor.description is not None:
-                columns = self.fetch_columns([(i[0], types_map.get(i[1], None)) for i in cursor.description])
-                rows = [dict(zip((c['name'] for c in columns), row)) for row in data]
-
+            queries = query.rstrip(';').split(';')
```

Comment:
That's not such a good idea, think of the following (contrived) example:

```
>>> "SELECT ';'".split(';')
["SELECT '", "'"]
```

But see the fix here: https://github.com/getredash/redash/pull/1955/commits/e02fdb3e37e5b6fefab591a44c74ecf4ce3b6223 for supporting multiple queries.

Once you rebase with latest, you can simplify the logic you implemented here.